app/services/vendor_api/generate_test_applications_for_provider.rb

```diff
@@ -12,16 +12,42 @@ def call(provider:, courses_per_application:, count:, for_ratified_courses: fals
 
   private
 
-    def course_list_for_provider(provider, for_ratified_courses)
+    def course_list_for_provider(provider, for_training_courses, for_ratified_courses, for_test_provider_courses)
+      if [for_training_courses, for_ratified_courses, for_test_provider_courses].none?
+        for_training_courses = true
+      end
+
+      course_lists = []
+
+      if for_training_courses
+        course_lists << courses_run_by_provider(provider)
+      end
       if for_ratified_courses
-        GetCoursesRatifiedByProvider.call(provider: provider)
-      else
-        Course.current_cycle
-              .open_on_apply
-              .joins(:course_options)
-              .distinct
-              .where(provider: provider)
+        course_lists << courses_ratified_by_provider(provider)
+      end
+      if for_test_provider_courses
+        course_lists << courses_run_by_test_provider
+      end
+
+      3.times.map do |index|
+        course_lists[index.modulo(course_lists.count)].sample
       end
```

Comment:
This logic needs to change. We pretty much want to get courses from the lists we have specified with the booleans, and then fill up the array (to length 3) with random courses from all available lists